From df7f7c50045475522d3716c642195e79ea6bc28d Mon Sep 17 00:00:00 2001
From: Michal Simek <michal.simek@xilinx.com>
Date: Thu, 8 Feb 2018 11:54:25 +0100
Subject: [PATCH 1/2] arm64: zynqmp: Enable newer pmufw versions

As of now newer pmufw is keeping old interfaces. That's why permit
Linux to run on newer version. Recommended version will be setup later.

And also show actual fw version not minimal one.

Signed-off-by: Michal Simek <michal.simek@xilinx.com>
Signed-off-by: Liu Haitao <haitao.liu@windriver.com>
Signed-off-by: Zhixiong Chi <zhixiong.chi@windriver.com>
---
 drivers/soc/xilinx/zynqmp/firmware.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/soc/xilinx/zynqmp/firmware.c b/drivers/soc/xilinx/zynqmp/firmware.c
index cc95e23ea77c..046919ff1ba5 100644
--- a/drivers/soc/xilinx/zynqmp/firmware.c
+++ b/drivers/soc/xilinx/zynqmp/firmware.c
@@ -446,7 +446,7 @@ static int __init zynqmp_plat_init(void)
 
 	/* Check PM API version number */
 	zynqmp_pm_get_api_version(&pm_api_version);
-	if (pm_api_version != ZYNQMP_PM_VERSION) {
+	if (pm_api_version < ZYNQMP_PM_VERSION) {
 		panic("%s power management API version error. Expected: v%d.%d - Found: v%d.%d\n",
 		       __func__,
 		       ZYNQMP_PM_VERSION_MAJOR, ZYNQMP_PM_VERSION_MINOR,
@@ -454,7 +454,7 @@ static int __init zynqmp_plat_init(void)
 	}
 
 	pr_info("%s Power management API v%d.%d\n", __func__,
-		ZYNQMP_PM_VERSION_MAJOR, ZYNQMP_PM_VERSION_MINOR);
+		pm_api_version >> 16, pm_api_version & 0xffff);
 
 	of_node_put(np);
 	return ret;
-- 
2.17.1

